plugin/src/main/java/io/jenkins/plugins/remotingkafka/KafkaCloudSlave.java

```diff
@@ -0,0 +1,128 @@
+package io.jenkins.plugins.remotingkafka;
+
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Descriptor;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.slaves.*;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.commons.lang.StringUtils;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class KafkaCloudSlave extends AbstractCloudSlave {
```

Comment:
IIUC I think the name slave is deprecated, we should use the name "agent" btw.
cc @oleg-nenashev can give you more context about this